From: Yves-Alexis Perez <corsac@debian.org>
Date: Thu, 3 Jan 2019 11:00:47 +0100
Subject: Fallback to XDG_SESSION_ID if sd_pid_get_session fails

---
 src/gs-listener-dbus.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/gs-listener-dbus.c b/src/gs-listener-dbus.c
index 1fe952c..e0ba004 100644
--- a/src/gs-listener-dbus.c
+++ b/src/gs-listener-dbus.c
@@ -2204,6 +2204,11 @@ init_session_id (GSListener *listener)
 #ifdef WITH_SYSTEMD
         g_free (listener->priv->sd_session_id);
         listener->priv->sd_session_id = query_sd_session_id (listener);
+        if (listener->priv->sd_session_id == NULL)
+        {
+                gs_debug ("Falling back to XDG_SESSION_ID environment variable");
+                listener->priv->sd_session_id = g_strdup(getenv("XDG_SESSION_ID"));
+        }
         gs_debug ("Got sd-session-id: %s", listener->priv->sd_session_id);
 #endif
 }
