From: Yves-Alexis Perez <corsac@debian.org>
Date: Thu, 3 Jan 2019 11:54:28 +0100
Subject: Print an error if session_id is NULL

---
 src/gs-listener-dbus.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/src/gs-listener-dbus.c b/src/gs-listener-dbus.c
index e0ba004..129f901 100644
--- a/src/gs-listener-dbus.c
+++ b/src/gs-listener-dbus.c
@@ -2199,7 +2199,10 @@ init_session_id (GSListener *listener)
 {
         g_free (listener->priv->session_id);
         listener->priv->session_id = query_session_id (listener);
-        gs_debug ("Got session-id: %s", listener->priv->session_id);
+        if (listener->priv->session_id == NULL)
+                g_error ("session_id is not set, is /proc mounted with hidepid>0?");
+        else
+                gs_debug ("Got session-id: %s", listener->priv->session_id);
 
 #ifdef WITH_SYSTEMD
         g_free (listener->priv->sd_session_id);
